libclamav/readdb.c

```diff
@@ -581,6 +581,106 @@ cl_error_t readdb_parse_ldb_subsignature(struct cli_matcher *root, const char *v
     return status;
 }
 
+static cl_error_t readdb_parse_yara_string(struct cli_matcher *root, const char *virname, char *hexsig, uint8_t subsig_opts,
+                                           const char *offset, uint8_t target, const uint32_t *lsigid, unsigned int options)
+{
+    cl_error_t status = CL_EPARSE;
+    cl_error_t ret;
+    char *hexcpy = NULL;
+
+    char *start = NULL, *end = NULL;
+
+    if (strchr(hexsig, '/')) {
```

Comment:
Suggest reversing `if/else` to de-nest some of the logic, or move the "pcre subsignature" handler into its own function to make the logic easier to follow.